### DISCUSSION TITLE
HybridSPAdes - On which stage of the pipeline are the graph outputs generated?

### DISCUSSION THREAD

**Discussion by nettanetta** [ablab/spades#923] [Q&A]
Hello!
I ran hybridSPAdes and I'm now looking at it's graph outputs (rather than the contigs or scaffolds). I'm currently using the assembly_graph.fastg because it was the easiest to parse and use for my needs. However, I do find that it's missing some information that I was able to extract when I used the contigs.

My main question is - on which stage are each of the graph outputs generated (assembly_graph.fastg, assembly_graph_after_simplification.gfa ,assembly_graph_with_scaffolds.gfa)? I couldn't find the full answer in the paper or in the readme.

To make it easier to answer, I'll try to state the main stages as I understood from the paper (with some minor question regarding these stages):
1. Constructing De Bruijn Graph - using short reads data only
2. Transforming the graph into an assembly graph - removing bulges, tips and chimeric edges (still using only short reads data, right?)
3. Mapping long reads to the assembly graph and generating read paths (are these read paths saved as an output?)
4. close gaps in the assembly graph using the consensus of long reads that span the gaps - are these included in any of the graph output files mentioned above?
5. Resolve repeats in the assembly graph by incorporating long read-paths into the decision rule of exSPAnder

I'll be happy to know when are the different graph outputs produced in order to better understand the type of information I should expect them to include. Also, If I misunderstood the stages, please correct me.

Thank you very much!
Netta

**Comment by asl** [comment]
Hello

All these files are generated from the same graph. And to simplify the things, all of them are saved after 4. The only difference is `assembly_graph_after_simplification.gfa` and `assembly_graph_with_scaffolds.gfa` – the latter also contains scaffold paths.

Note that all these steps you mentioned above is quite a simplification. The underlying ideas are the same, yes, however, many things are much more complex – the paper is not enough to discuss all these details here and there.

PS: No need to "parse" anything by hands: there are lots of parsers for GFA, e.g. GfaPy (for Python) or gfatools for C/C++.
